exp/views/study.py

```diff
@@ -905,6 +905,18 @@ def dispatch(self, request, *args, **kwargs):
             return HttpResponseRedirect(get_external_url(study, response))
         else:
             _, _, _, study_uuid, _, _, _, *rest = request.path.split("/")
+            """If locale (language code) is present in the URL, then there will be an extra element
+            returned from split("/") before the study UUID, so the UUID variable will be "studies".
+            In this case we need to add an extra returned string before study_uuid to capture the values correctly,
+            and then re-write the request path so that it points to a working study URL.
+            """
+            if study_uuid == "studies":
+                _, _, exp, studies, study_uuid, child_uuid, _, _, *rest = request.path.split("/")
+                path_no_locale = "/"+"/".join([exp, studies, study_uuid, child_uuid, "preview/"])
+                request.path = path_no_locale
+                request.path_info = path_no_locale
+                request.META['HTTP_REFERER'] = request.META["BASE_URL"] + path_no_locale
+                request.META['PATH_INFO'] = path_no_locale
```

Comment:
This is bit of a finicky suggestion, but could we remove the duplicate code?

Reply:
🤦‍♀️  Not finicky! Will do.

Reply:
@okaycj I removed the change to `HTTP_REFERER` because it was uncessary. But I think the changes to the request `path`, `path_info` and `META.PATH_INFO` are all necessary. (It's hard to know for sure because the study links produce an error in development anyway - I think I'll have to test on staging).

So I don't think there's code duplication here, unless I've misunderstood?